views/school_experience.sql

```diff
@@ -0,0 +1,139 @@
+alter view git.school_experience_events as
+    -- shows each 'event' that happens in the lifecycle of a school
+    -- experience request
+    --
+    -- we need to do some cleaning up before we actually select any data
+    -- so the view definition's split into three bits:
+    --
+    --
+    -- splitting the log:
+    --
+    -- The school experience application writes data to a single varchar(max)
+    -- column (dfe_notesforclassroomexperience), probably Dynamics-related
+    -- (GITIS was still pretty new when SE was integrated)
+    --
+    -- The format is a header line, followed by a blank line, then rows
+    -- of data:
+    --
+    -- RECORDED   ACTION                 EXP DATE   URN    NAME
+    -- 01/01/2021 REQUEST                01/02/2021 111111 Springfield Elementary School
+    -- 02/01/2021 ACCEPTED               01/02/2021 111111 Springfield Elementary School
+    --
+    -- When there's no data the string 'RTBF' is present; not sure what this means
+    -- Some events have some lines of raw text that follow it, they are notes about
+    -- the event and we ignore them here.
+    with split_log as (
+        select
+            id,
+            value as line
+    
+        from
+            crm_contact
+            -- use cross apply with string_split to yield a row in the resultset PER
+            -- row in the unstructured data - split on newline
+            cross apply string_split(
+                -- make things easier by replacing the \r\n with just \n
+                replace(
+                    dfe_notesforclassroomexperience,
+                    char(13)+char(10),
+                    char(13)
+                ), char(13)
+            )
+
+        where
+            dfe_notesforclassroomexperience is not null  -- only target SE-related rows
+            and value <> 'RTBF'                          -- no idea what RTBF stands for but looks like 'no data'!
+            and value <> 'none'                          -- no data (why are these here?)
+            and value not like 'RECORDED%'               -- not the header row
+            and len(trim(value)) > 1                     -- not the empty row beneath the header
```

Comment:
These criteria aren't really necessary now the date 'plucking' is in, but they at least provide a bit of context on the other stuff that's crammed in this column so I left them in.